PERL/ASGSUtil.pm

```diff
@@ -103,6 +103,27 @@ sub setParameter {
    }
    return;
 }
+
+# create a hash of properties from run.properties
+sub readProperties {
+   my ( $runProp_ref, $fileName ) = @_;
+   # open properties file
+   my $rp;
+   unless ( open($rp,"<","$fileName") ) {
```

Comment:
I find regular `if` more readable, the `unless` idiom tends to cause confusion when sprinkled about - though I have seem places where it works. If it's all the same to you, can you convert this to,

   if (not open(my $rp, "<", $fileName)) {
       stderrMessage("ERROR", "Failed to open '$fileName': $!.");
       return
   }
   
   I am pretty sure that in this case the `my $rp` retains scope (`if` doesn't create new lexical scope) so you don't have to do the `my $rp` first.

Reply:
Can do